src/realizations/catchment/Realization_Config.cpp

```diff
@@ -0,0 +1,291 @@
+                message += ", ";
+            }
+        }
+        
+        throw std::runtime_error(message);
+    }
+
+    std::string path = this->forcing_parameters.at("path").as_string();
+
+    if (this->forcing_parameters.count("file_pattern") == 0) {
+        return forcing_params(
+            path,
+            this->forcing_parameters.at("start_time").as_string(),
+            this->forcing_parameters.at("end_time").as_string()
+        );
+    }
+
+    // Since we are given a pattern, we need to identify the directory and pull out anything that matches the pattern
+    if (path.compare(path.size() - 1, 1, "/") != 0) {
+        path += "/";
+    }
+
+    std::string filepattern = this->forcing_parameters.at("file_pattern").as_string();
+
+    int id_index = filepattern.find("{{ID}}");
+
+    // If an index for '{{ID}}' was found, we can count on that being where the id for this realization can be found.
+    //     For instance, if we have a pattern of '.*{{ID}}_14_15.csv' and this is named 'cat-87',
+    //     this will match on 'stuff_example_cat-87_14_15.csv'
+    if (id_index != std::string::npos) {
+        filepattern = filepattern.replace(id_index, sizeof("{{ID}}") - 1, this->id);
+    }
+
+    // Create a regular expression used to identify proper file names
+    std::regex pattern(filepattern);
+
+    // A stream providing the functions necessary for evaluating a directory: 
+    //    https://www.gnu.org/software/libc/manual/html_node/Opening-a-Directory.html#Opening-a-Directory
+    DIR *directory = nullptr;
+
+    // structure representing the member of a directory: https://www.gnu.org/software/libc/manual/html_node/Directory-Entries.html
+    struct dirent *entry = nullptr;
+
+    // Attempt to open the directory for evaluation
+    directory = opendir(path.c_str());
+
+    // If the directory could be found, we can go ahead and iterate
+    if (directory != nullptr) {
+        while ((entry = readdir(directory))) {
+            // If the entry is a regular file or symlink AND the name matches the pattern, 
+            //    we can consider this ready to be interpretted as valid forcing data (even if it isn't)
+            if ((entry->d_type == DT_REG or entry->d_type == DT_LNK) and std::regex_match(entry->d_name, pattern)) {
+                return forcing_params(
+                    path + entry->d_name,
+                    this->forcing_parameters.at("start_time").as_string(),
+                    this->forcing_parameters.at("end_time").as_string()
+                );
+            }
+        }
+    }
+    else {
+        // The directory wasn't found; forcing data cannot be retrieved
+        throw std::runtime_error("No directory for forcing data was found at: " + path);
+    }
+
+    closedir(directory);
+
+    throw std::runtime_error("Forcing data could not be found for '" + this->id + "'");
+}
+
+double Realization_Config_Base::get_response(double input_flux, time_step_t t, time_step_t dt, void* et_params) {
```

Comment:
Not sure this redirect is required.  The caller could probably do a `get_realization().get_response` just as well.  This just happens to be the third redirect to a call to run the model, starting to wonder how many is enough haha.

Reply:
I actually don't think we have enough redirects! (That's a joke)

`get_realization().get_response` isn't bad. I default to the law of demeter, which is why the config itself has the function. Both work.